script/helperFunctions.sh

```
@@ -2011,47 +2010,84 @@ function verifyContract() {
+      echo "[info] Verification still pending after --watch, waiting 30 seconds before checking final status..."
+      sleep 30
+
+      # Check final status
+      local CHECK_CMD=()
+      if isZkEvmNetwork "$NETWORK"; then
+        CHECK_CMD=("./foundry-zksync/forge" "verify-check" "--zksync" "--chain" "$CHAIN_ID" "$ADDRESS")
+      else
+        CHECK_CMD=("forge" "verify-check" "--chain" "$CHAIN_ID" "$ADDRESS")
+      fi
+
+      local CHECK_OUTPUT=$(FOUNDRY_LOG=trace "${CHECK_CMD[@]}" 2>&1)
```

Shellcheck (0.10.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)